services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/model/CloudFrontSignerRequest.java

```diff
@@ -0,0 +1,217 @@
+        this.keyPairId = builder.keyPairId;
+        this.expirationDate = builder.expirationDate;
+        this.activeDate = builder.activeDate;
+        this.ipRange = builder.ipRange;
+    }
+
+    /**
+     * Create a builder that can be used to create a {@link CloudFrontSignerRequest}
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder(this);
+    }
+
+    /**
+     * Returns the resource URL
+     */
+    public String resourceUrl() {
+        return resourceUrl;
+    }
+
+    /**
+     * Returns the private key
+     */
+    public PrivateKey privateKey() {
+        return privateKey;
+    }
+
+    /**
+     * Returns the key pair ID
+     */
+    public String keyPairId() {
+        return keyPairId;
+    }
+
+    /**
+     * Returns the expiration date
+     */
+    public Instant expirationDate() {
+        return expirationDate;
+    }
+
+    /**
+     * Returns the active date
+     */
+    public Instant activeDate() {
+        return activeDate;
+    }
+
+    /**
+     * Returns the IP range
+     */
+    public String ipRange() {
+        return ipRange;
+    }
+
+    @NotThreadSafe
+    @SdkPublicApi
+    public interface Builder extends CopyableBuilder<CloudFrontSignerRequest.Builder, CloudFrontSignerRequest> {
+
+        /**
+         * Configure the resource URL to be signed
+         * <p>
+         *     The URL or path that uniquely identifies a resource within a
+         *     distribution. For standard distributions the resource URL will
+         *     be <tt>"http://" + distributionName + "/" + objectKey</tt>
+         *     (may also include URL parameters. For distributions with the
+         *     HTTPS required protocol, the resource URL must start with
+         *     <tt>"https://"</tt>
+         * </p>
+         */
+        Builder resourceUrl(String resourceUrl);
+
+        /**
+         * Configure the private key to be used to sign the policy
+         */
+        Builder privateKey(PrivateKey privateKey);
+
+        /**
+         * Configure the ID of the key pair stored in the AWS account
+         */
+        Builder keyPairId(String keyPairId);
+
+        /**
+         * Configure the expiration date of the signed URL or signed cookie
+         */
+        Builder expirationDate(Instant expirationDate);
+
+        /**
+         * Configure the active date of the signed URL or signed cookie - for custom policies
+         */
+        Builder activeDate(Instant activeDate);
+
+        /**
+         * Configure the IP range of the signed URL or signed cookie - for custom policies (optional field)
+         * <p>
+         *     The allowed IP address range of the client making the GET
+         *     request, in CIDR form (e.g. 192.168.0.1/24).
+         * </p>
+         */
+        Builder ipRange(String ipRange);
+    }
+
+    @SdkInternalApi
```

Comment:
Nit: Not needed, having it on the main class is enough

Reply:
removed